### PR TITLE
Fix #1

### LINKED ISSUE #1: log level never considered under INFO level.
Whenever I try to lower the logging level from INFO (20), I'm never able to printout messages passed to logdebug, logfine, etc...

test case:

``` R
> library(logging)
> basicConfig("DEBUG")  # or something lower than 20
> getLogger()
Reference class object of class "Logger"
Field "name":
[1] ""
Field "handlers":
$basic.stdout
<environment: 0x9e23c8c>

Field "level":
DEBUG 
   10 
> loginfo("info")
2013-02-07 11:48:29 INFO::info # ok, as I expected...
> logdebug("debug") # ... shouldn't I see the debug message here?
> logReset() # let's see if at higher levels works
> basicConfig("ERROR")
> getLogger()
Reference class object of class "Logger"
Field "name":
[1] ""
Field "handlers":
$basic.stdout
<environment: 0x8d64934>

Field "level":
ERROR 
   40 
> loginfo("info")
> logwarn("warn")
> logerror("error")
2013-02-07 11:50:05 ERROR::error # no problems here.
```

The version I'm using through CRAN is 0.6-92, maybe I should update from here ;)

HTH, 
Giuseppe

### PR DESCRIPTION
**Pull request by giupo**
Hi, 

I think I could have fixed the issue I opened lately, (#1). All tests pass: let me know if I can help!

If it's ok with it, I (you have to ;) ) would update it on CRAN...

Ciao!
Giuseppe

### CODE REVIEW

File: pkg/R/logger.R [hunk 1]
```diff
@@ -93,18 +93,18 @@ getLogger <- function(name='', ...)
...
-  }
+  } 
   logging.options[[fullname]]
 }
 
 basicConfig <- function(level=20) {
   rootLogger <- getLogger()
   updateOptions(rootLogger, level=namedLevel(level))
-  rootLogger$addHandler('basic.stdout', writeToConsole)
+  rootLogger$addHandler('basic.stdout', writeToConsole, level=level)
```
Comment (giupo) [inline pkg/R/logger.R:107]: This is what fixes the problem (for me, if I've understood how it works...)

**Comment by krlmlr** [comment]
Unfortunately this breaks backward compatibility. I have sent a patch to Mario (and also attached it to the corresponding R-Forge issue) that adds a new parameter to `basicConfig()`.

**Comment by mfrasca** [comment]
good day Giuseppe and Kirill,
as I wrote to Kirill, I am afraid this would break backwards compatibility.
but while answering to Giuseppe, I formulated my answer in such a way that I saw that it was too complicated... too complicated usually means there's a mistake somewhere. so I reconsidered my answer to Kirill...
the simple formulation I could see is this:
- you initialize the system with `basicConfig`, this creates a handler and attaches it to the root logger.
- you use any logger you may need, knowing that all your log-records will be collected by the handler attached at the root.
- altering the level of the logger is something that the logging code should do (possibly a package).
- altering the level of the handler is something you do when configuring the logging system (possibly in a script).
- if you are in a hurry and you use `basicConfig`, things should "just work".
  so I think I agree with both of you when you request a change in the current behaviour.

**Comment by mfrasca** [comment]
hi Giuseppe, but this #2 is not an issue... the issue is #1 
I am closing this as invalid, and I'm sorry for this...
but anyway, it's a chance for you to add some unit tests to your pull request!! :)

p.s.: oh, by the way, the patch does not break any previous test, as this part is apparently not well tested at all!

**Comment by giupo** [comment]
Hi Mario, 

This is just my inexperience to git/github ;). Sorry for the burden ;)

Giuseppe

**Pull request closed without merging**